include/chunk/region.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "chunk/chunk.h"
+#include <memory>
+
+namespace chunk
+{
+
+/**
+ * Class containing hash function for std::unordered_map
+ */
+
+namespace detail
+{
+struct ChunkHasher
+{
+    std::size_t operator()( const pos::ChunkPos& chunk_pos ) const
+    {
+        std::hash<int64_t> hasher{};
+        return hasher( int64_t( chunk_pos ) );
+    }
+};
+} // namespace detail
+
+/**
+ * Class that contains all block ides and manage the chunks
+ */
+
+class Region
+{
+  public:
+    using ChunkMap = std::unordered_map<pos::ChunkPos, Chunk, detail::ChunkHasher>;
+    using BlockIDPtr = std::unique_ptr<BlockID[]>;
+
+  public:
+    Region( const Region& ) = delete;
+    Region( Region&& ) = delete;
+
+    Region& operator=( const Region& ) = delete;
+    Region& operator=( Region&& ) = delete;
```

Comment:
Why is this class not movable? 
Copy constructors are implicitly deleted because of non-copyable member ```std::unique_ptr```.

Reply:
Why not? This is singleton object, copy and even move operations are not valid in my opinion

Reply:
Touche.
Although this this can be achieved by making all constructors ```private```